# $NetBSD$
#
###########################################################
#                  Generated by fbsd2pkg                  #
#              Thu Jul 20 18:34:44 CDT 2017               #
###########################################################

DISTNAME=	canu-${PORTVERSION}
CATEGORIES=	biology java
MASTER_SITES=	${MASTER_SITE_GITHUB:=marbl/}
GITHUB_PROJECT=	canu
GITHUB_TAG=	v${PORTVERSION}

MAINTAINER=	bacon4000@gmail.com
HOMEPAGE=	http://canu.readthedocs.io/
COMMENT=	Single molecule sequence assembler for genomes large and small

# Check this
LICENSE=	gnu-gpl-v2

# Too heavy.  Disable check in canu command instead.
# DEPENDS+=	gnuplot>0:../../graphics/gnuplot

# Screwy Makefile compiles directly into ${DESTDIR}${PREFIX}
SUBST_CLASSES+=		optimize
SUBST_STAGE.optimize=	post-patch
SUBST_SED.optimize+=	-e 's|-O4|-O2|g'
SUBST_SED.optimize+=	-e 's|-funroll-loops||g'
SUBST_SED.optimize+=	-e 's|-fexpensive-optimizations||g'
SUBST_FILES.optimize+=	${WRKSRC}/Makefile

SUBST_CLASSES+=		jar
SUBST_STAGE.jar=	post-patch
SUBST_SED.jar=		-e 's|\\$$bin/mhap-|${PREFIX}/${JAVAJARDIR}/mhap-|g'
SUBST_FILES.jar=	${WRKSRC}/pipelines/canu/OverlapMhap.pm

SUBST_CLASSES+=		perl
SUBST_STAGE.perl=	post-patch
SUBST_SED.perl=		-e 's|RealBin/lib|RealBin/../${SITE_PERL_REL}|g'
SUBST_FILES.perl=	${WRKSRC}/pipelines/canu.pl

REPLACE_PERL+=	bogart/plotErrorProfile.pl \
		bogart/findOverlappingReads.pl \
		pipelines/simple-repeat-test.pl \
		pipelines/canu-object-store.pl \
		pipelines/canu.pl \
		pipelines/bogart-sweep.pl \
		bogus/bogusness-run.pl \
		bogart-analysis/analyze-nucmer-gaps.pl \
		bogart-analysis/examine-mapping.pl \
		bogart-analysis/bogart-build-fasta.pl \
		bogart-analysis/examine-mapping-ideal.pl \
		bogart-analysis/count-bubbles.pl \
		bogart-analysis/locate-read-in-unitig-based-on-olaps.pl \
		bogart-analysis/analyze-mapped-unitigs-for-joins.pl \
		bogart-analysis/show-false-best-edges-from-mapping.pl \
		overlapBasedTrimming/detect-unsplit-subreads-in-overlaps.pl \
		overlapBasedTrimming/test-random-chimeric-fragments.pl \
		overlapBasedTrimming/generate-random-chimeric-fragments.pl \
		overlapInCore-analysis/infer-ovl-from-genomic-blasr.pl \
		overlapInCore-analysis/infer-obt-from-genomic-blasr.pl \
		overlapInCore-analysis/infer-olaps-from-pairwise-coords.pl \
		overlapInCore-analysis/find-missed-true-overlaps.pl \
		overlapInCore-analysis/infer-olaps-from-genomic-coords.pl \
		overlapInCore-analysis/analyze-true-vs-test.pl \
		overlapInCore-analysis/filterTrue.pl \
		overlapInCore-analysis/infer-olaps-from-pairwise-blasr.pl \
		overlapInCore-analysis/check-ordered-reads-for-missed-overlaps.pl \
		merTrim/merTrim-compare-logs.pl \
		erateEstimate/erate-estimate-test-based-on-mapping.pl \
		erateEstimate/erate-estimate-plot-per-base-estimate.pl \
		canu_version_update.pl \
		meryl/gkrpt.pl \
		fastq-utilities/fastqSimulate-checkCoverage.pl \
		fastq-utilities/fastqSimulate-perfectSep.pl \
		addCopyrights-BuildData.pl \
		addCopyrights.pl

# May work on other 64-bit processors, but untested
ONLY_FOR_PLATFORM=	 *-*-x86_64

USE_LANGUAGES=	c c++
USE_JAVA=	run
USE_TOOLS+=	gmake pax perl

GCC_REQD+=	4.8

WRKSRC=		${WRKDIR}/canu-${PORTVERSION}/src

MAKE_ENV+=	DESTDIR=${WRKSRC}

PORTVERSION=	1.5

TMP_INST=	${WRKSRC}${PREFIX}/${OPSYS}-${MACHINE_ARCH:S/x86_64/amd64/}
SITE_PERL_REL=	lib/perl5/site_perl
JAVAJARDIR=	share/java/classes
INSTALLATION_DIRS=	bin \
			${SITE_PERL_REL} \
			${JAVAJARDIR}

post-build:
	${MKDIR} ${TMP_INST}/${SITE_PERL_REL}
	${MV} ${TMP_INST}/bin/lib/canu ${TMP_INST}/${SITE_PERL_REL}
	${RMDIR} ${TMP_INST}/bin/lib
	${MV} ${TMP_INST}/bin/*.a ${TMP_INST}/lib
	${MKDIR} ${TMP_INST}/${JAVAJARDIR}
	${MV} ${TMP_INST}/bin/*.jar ${TMP_INST}/${JAVAJARDIR}

do-install:
	cd ${TMP_INST}/bin && ${PAX} -wr * ${DESTDIR}${PREFIX}/bin
	cd ${TMP_INST}/lib && ${PAX} -wr * ${DESTDIR}${PREFIX}/lib
	cd ${TMP_INST}/share && ${PAX} -wr * ${DESTDIR}${PREFIX}/share

.include "../../devel/boost-libs/buildlink3.mk"
.include "../../lang/perl5/module.mk"
.include "../../mk/bsd.pkg.mk"
